docs/api-docs/analytics/bodl-overview.md

```diff
@@ -0,0 +1,427 @@
+| cartItemAdded | [{{cart.added_item}}](/theme-objects/carts) |
+| categoryProducts | [{{category.products}}](/theme-objects/category) |
+| categoryName | [{{category.name}}](/theme-objects/category) |
+| order | [{{order}}](/theme-objects/order) |
+| productId | [{{product.id}}](/theme-objects/product) |
+| productTitle | [{{product.title}}](/theme-objects/product) |
+| productCurrency | [{{product.price.without_tax.currency}}](/theme-objects/product) |
+| productPrice | [{{product.price.without_tax.value}}](/theme-objects/product) |
+| products | [{{products}}](/theme-objects/products) |
+| search | [{{product_results}}](/theme-objects/product_results) |
+| wishlist | [{{wishlist}}](/theme-objects/wishlist) |
+
+
+A standard `BODL` instance contains the following calculated properties:
+
+| BODL Property | First-Level Child Properties | 
+|:---|:---|
+| category: object | name: string, products: array |
+| product: object | id: string, title: string, price: object |
+| customer: object |  |
+
+A standard `BODL` instance contains the following methods:
+
+| BODL Method | Parameter(s) | Parameter Properties | Return Value |
+|:---|:---|:---|:---|
+| getCartItemContentId | `item`: object | `item.type`: string, `item.product_id`: string | If the item is a gift certificate, returns the item's `type`. Otherwise, returns the item's `product_id`.  |
+| getQueryParamValue | `name`: string | **X** | Returns the value of the specified query parameter from the window's URL. |
+
+## Initialize BODL
+
+The following script extracts storefront data from the Stencil objects available in the front-end environment to construct a standard `BODL` object.
+
+```handlebars title="Sample Script Code Start: Initialization Script & Page Event" lineNumbers
+<script>
+  if (typeof BODL === 'undefined') {
+    // https://developer.bigcommerce.com/theme-objects/breadcrumbs
+    {{inject "breadcrumbs" breadcrumbs}}
+    // https://developer.bigcommerce.com/theme-objects/brand
+    {{inject "brand" brand}}
+    // https://developer.bigcommerce.com/theme-objects/category
+    {{inject "categoryProducts" category.products}}
+    {{inject "categoryName" category.name}}
+    // https://developer.bigcommerce.com/theme-objects/product
+    {{inject "productId" product.id}}
+    {{inject "productTitle" product.title}}
+    {{inject "productCurrency" product.price.without_tax.currency}}
+    {{inject "productPrice" product.price.without_tax.value}}
+    // https://developer.bigcommerce.com/theme-objects/products
+    {{inject "products" products}}
+    // https://developer.bigcommerce.com/theme-objects/product_results
+    {{inject "search" product_results}}
+    // https://developer.bigcommerce.com/theme-objects/order
+    {{inject "order" order}}
+    // https://developer.bigcommerce.com/theme-objects/wishlist
+    {{inject "wishlist" wishlist}}
+    // https://developer.bigcommerce.com/theme-objects/cart_id
+    {{inject "cartId" cart_id}}
```

Comment:
@becomevocal will adding `cart_id` create a parse error?

Reply:
As long as you check for the existence before using it, I don't think so.